ebos/ecloutputblackoilmodule.hh

```diff
@@ -142,27 +145,32 @@ public:
                    getPropValue<TypeTag, Properties::EnableMICP>())
          , simulator_(simulator)
     {
-        const SummaryConfig summaryConfig = simulator_.vanguard().summaryConfig();
-
-        for (auto& region_pair : this->regions_)
-            createLocalRegion_(region_pair.second);
+        for (auto& region_pair : this->regions_) {
+            this->createLocalRegion_(region_pair.second);
+        }
 
-        for (const auto& node: summaryConfig) {
-            if (node.category() == SummaryConfigNode::Category::Block) {
-                if (collectToIORank.isCartIdxOnThisRank(node.number() - 1)) {
-                    std::pair<std::string, int> key = std::make_pair(node.keyword(), node.number());
-                    this->blockData_[key] = 0.0;
-                }
+        for (const auto& node : this->simulator_.vanguard().summaryConfig()) {
+            if ((node.category() == SummaryConfigNode::Category::Block) &&
+                collectToIORank.isCartIdxOnThisRank(node.number() - 1))
+            {
+                this->blockData_.emplace(std::piecewise_construct,
+                                         std::forward_as_tuple(node.keyword(),
+                                                               node.number()),
+                                         std::forward_as_tuple(0.0));
```

Comment:
Shouldn't 
```
this->blockData_.try_emplace( {node.keyword(), node.number()),
                                                  0.0);
```
work since c++17?

Reply:
> Shouldn't [`try_emplace()`] work since c++17?

Yes, but that doesn't do the same thing as `emplace(piecewise_construct, ...)`.  `Try_emplace()` with an initialiser list will create an instance of the key type, in this case a `pair<string, int>`, and then create a "node" by copying the key and the value into it.  On the other hand, `emplace(piecewise_construct, ...)` will create the node in place and not copy anything.